### PR TITLE
Use SparseArrayCompat.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
#### What is it?
- [x] Codebase improvement (dev facing)

#### Description of the changes in your PR

- Use [`SparseArrayCompat`](https://developer.android.com/reference/androidx/collection/SparseArrayCompat) to reduce memory usage in the thumbnail preview, search and download functionality by avoiding integer boxing.
- Make some additional minor code improvements in `StreamItemAdapter`.

#### Fixes the following issue(s)

- Fixes #

#### APK testing

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/util/StreamItemAdapter.java [hunk 1]
```diff
@@ -140,68 +120,41 @@ private View getCustomView(final int position,
...
-            }
-        }
-
-        if (streamsWrapper.getSizeInBytes(position) > 0) {
-            final SecondaryStreamHelper<U> secondary = secondaryStreams == null ? null
-                    : secondaryStreams.get(position);
-            if (secondary != null) {
-                final long size = secondary.getSizeInBytes()
-                        + streamsWrapper.getSizeInBytes(position);
-                sizeView.setText(Utility.formatBytes(size));
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/util/StreamItemAdapter.java:171]: `sizeView` is not being set anymore

File: app/src/main/java/org/schabi/newpipe/util/StreamItemAdapter.java [hunk 2]
```diff
@@ -140,68 +120,41 @@ private View getCustomView(final int position,
...
-            } else {
-                sizeView.setText(streamsWrapper.getFormattedSize(position));
-            }
-            sizeView.setVisibility(View.VISIBLE);
-        } else {
-            sizeView.setVisibility(View.GONE);
-        }
-
-        if (stream instanceof SubtitlesStream) {
-            formatNameView.setText(((SubtitlesStream) stream).getLanguageTag());
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/util/StreamItemAdapter.java:181]: `formatNameView` is not being set anymore

File: app/src/main/java/org/schabi/newpipe/player/seekbarpreview/SeekbarPreviewThumbnailHolder.java [hunk 3]
```diff
@@ -63,13 +59,12 @@ public synchronized void resetFrom(
...
-    private void resetFromAsync(
-            final int seekbarPreviewType,
-            final List<Frameset> framesets,
-            final UUID updateRequestIdentifier) {
-
+    private void resetFromAsync(final int seekbarPreviewType, final List<Frameset> framesets,
+                                final UUID updateRequestIdentifier) {
         Log.d(TAG, "Clearing seekbarPreviewData");
-        seekbarPreviewData.clear();
+        synchronized (seekbarPreviewData) {
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/player/seekbarpreview/SeekbarPreviewThumbnailHolder.java:65]: Why is `synchronized` now needed?

**Review by Stypox — CHANGES_REQUESTED** [review]
This PR is not just about `SparseArrayCompat`. It contains a lot of other refactorings. Please document them properly in the PR description, otherwise it's difficult to understand why you made some changes.

Have you tested these changes? This is what I'm getting...
image

**Comment by Isira-Seneviratne** [comment]
> This PR is not just about `SparseArrayCompat`. It contains a lot of other refactorings. Please document them properly in the PR description, otherwise it's difficult to understand why you made some changes.

Sorry about that, will do. 

> 
> Have you tested these changes? This is what I'm getting...
> 

My bad again, I'll restore the text. Thanks for the review!

File: app/src/main/java/org/schabi/newpipe/player/seekbarpreview/SeekbarPreviewThumbnailHolder.java (hunk 3 above)
Comment (Isira-Seneviratne) [reply app/src/main/java/org/schabi/newpipe/player/seekbarpreview/SeekbarPreviewThumbnailHolder.java:65]: Android's sparse arrays are not thread-safe.

**Review by Stypox — APPROVED** [review]
Thank you! I tested thoroughly and it works

**Pull request merged**